File: src/main/java/infobip/test/shortener/url/AddPathCommand.java
package infobip.test.shortener.url;

import infobip.test.shortener.model.UrlData;
import org.axonframework.commandhandling.annotation.TargetAggregateIdentifier;
import org.immutables.value.Value;

@Value.Immutable
public interface AddPathCommand {

    @TargetAggregateIdentifier
    String getPath();
    UrlData getData();
}

File: src/main/java/infobip/test/shortener/url/AddPathCommand.kt
package infobip.test.shortener.url

import infobip.test.shortener.model.UrlData
import org.axonframework.commandhandling.annotation.TargetAggregateIdentifier

data class AddPathCommand(@TargetAggregateIdentifier val path: String? = null,
                          val data: UrlData? = null)